src/control/BarrettFingerKinematicSimulationSpreadCommandExecutor.cpp

```diff
@@ -104,53 +101,53 @@ std::future<void> BarrettFingerSpreadCommandExecutor::execute(
 
     mPromise.reset(new std::promise<void>());
 
-    if (_goalPosition < mDofLimits.first)
+    if (goalPosition < mDofLimits.first)
       mGoalPosition = mDofLimits.first;
-    else if (_goalPosition > mDofLimits.second)
+    else if (goalPosition > mDofLimits.second)
       mGoalPosition = mDofLimits.second;
     else
-      mGoalPosition = _goalPosition;
+      mGoalPosition = goalPosition;
 
-    mCollideWith = _collideWith;
     mInExecution = true;
 
     return mPromise->get_future();
   }
 }
 
 //=============================================================================
-void BarrettFingerSpreadCommandExecutor::step(double _timeSincePreviousCall)
+void BarrettFingerKinematicSimulationSpreadCommandExecutor::step(
+  const std::chrono::milliseconds& timeSincePreviousCall)
 {
-  using std::chrono::milliseconds; 
+  auto period = std::chrono::duration<double>(
+    timeSincePreviousCall).count();
 
   // Terminate the thread if mRunning is false.
   std::lock_guard<std::mutex> lock(mMutex);
 
   if (!mInExecution)
     return;
-  
+
   // Current spread. Check that all spreads have same values.
   double spread = mSpreadDofs[0]->getPosition();
-  for(int i=1; i < kNumFingers; ++i)
+  for(size_t i = 1; i < kNumFingers; ++i)
```

Comment:
I don't think it's off-by-one. It uses the 0th-finger's spread value to check that the rest of the fingers (the other finger) has the same spread value. We implemented this because the spread is symmetric.